### ISSUE TITLE
No way to reset query in model

### ISSUE THREAD

**Issue by nebulousGirl** [TypeRocket/typerocket#162]
I had to make multiple queries in a model to find duplicates that I needed to update. The problem is that the where clause is only incremental, I cannot effectively reset the query to "SELECT * FROM table". 

I have bypassed this problem by creating a new instance of the query class, but I think that is a flaw in the model/query class.

**Comment by nebulousGirl** [comment]
Methods like  `findById` and `findAll` should reset the where clause I think.

**Comment by brettvan** [comment]
I just ran into this one myself where I was pulling ACL rules from a table where I did multiple passes to select all the rules for a resource based on various roles a logged in user belonged to.  Could I ask what were the changes you made to the query class?

As a test, I added  

unset($this->query['where']);

at the bottom of the runQuery() method just to see if that was all it was.  Yes, that seemed to fix it.
